### PR TITLE
Run JuliaFormatter, add JuliaFormatter workflow

### PR DESCRIPTION
**Pull request by sethaxen**
This essentially adds a linter to CI.

### CODE REVIEW

**Comment by mateuszbaran** [comment]
What does it actually do? It just fails the test if formatting makes any changes?

**Comment by sethaxen** [comment]
> What does it actually do? It just fails the test if formatting makes any changes?

That's right, yeah. And it runs it on the entire repo, not just the source code. It's not included in the CI workflow, so its status won't impact our badges; it's mostly there as a reminder to us before we merge PRs, but I enabled it for tags and master as for the CI.

**Comment by sethaxen** [comment]
I'm also going to use this PR to troubleshoot why nightly failures are not tolerated.

**Comment by sethaxen** [comment]
The current failure is due to JuliaFormatter not being idempotent on our `@invoke_maker` calls. Relevant issue: https://github.com/domluna/JuliaFormatter.jl/issues/218

File: src/groups/group.jl [hunk 1]
```diff
@@ -225,25 +230,29 @@ function get_vector(M::AbstractGroupManifold, e::Identity, X, B::VeeOrthogonalBa
     return get_vector(decorated_manifold(M), e.p, X, B)
 end
 function get_vector(M::Manifold, e::Identity, X, B::VeeOrthogonalBasis)
-    M != e.group.manifold && error("On $(M) the identity $(e) does not match to perform get_vector.")
+    M != e.group.manifold &&
+    error("On $(M) the identity $(e) does not match to perform get_vector.")
```
Comment (mateuszbaran) [inline src/groups/group.jl:234]: This is one thing the formatter does I don't like: breaking `condition && something()` lines without indenting. Maybe we should just replace them with `if` in cases where they are too long?

Comment (sethaxen) [reply src/groups/group.jl:234]: Ah, yes, I think we've agreed before that way's better. Ideally in the future with the workflow, we'll all run JuliaFormatter before merging, and large runs like this where it's easy to miss these won't be necessary. If you open an issue on JuliaFormatter, they might agree and change the default.

Comment (sethaxen) [reply src/groups/group.jl:234]: I created an issue for this: https://github.com/domluna/JuliaFormatter.jl/issues/222

Comment (mateuszbaran) [reply src/groups/group.jl:234]: Thanks!

**Pull request merged**